### PR TITLE
Fix delete! for WeakKeyDict.

### PR DESCRIPTION
**Pull request by twavv**
Relatively minor thing. Currently, `delete!` for `WeakKeyDict`s returns the underlying dict implementation rather than the `WeakKeyDict` like it's supposed to.

```julia
# Julia 1.3
julia> wkd = WeakKeyDict()
WeakKeyDict{Any,Any} with 0 entries

julia> delete!(wkd, "foo")
Dict{WeakRef,Any} with 0 entries
```

### CODE REVIEW

File: base/weakkeydict.jl [hunk 1]
```diff
@@ -105,7 +105,7 @@ function get!(default::Callable, wkh::WeakKeyDict{K}, key) where {K}
 end
 pop!(wkh::WeakKeyDict{K}, key) where {K} = lock(() -> pop!(wkh.ht, key), wkh)
 pop!(wkh::WeakKeyDict{K}, key, default) where {K} = lock(() -> pop!(wkh.ht, key, default), wkh)
-delete!(wkh::WeakKeyDict, key) = lock(() -> delete!(wkh.ht, key), wkh)
+delete!(wkh::WeakKeyDict, key) = (lock(() -> delete!(wkh.ht, key), wkh); whk)
```
Comment (JeffBezanson) [inline base/weakkeydict.jl]:
```suggestion
delete!(wkh::WeakKeyDict, key) = (lock(() -> delete!(wkh.ht, key), wkh); wkh)
```

**Comment by JeffBezanson** [comment]
Good catch!

**Comment by travigd** [comment]
Fixed typo 🙃

**Comment by travigd** [comment]
Happy to add to NEWS.md, but not sure where this would fit in.

**Comment by KristofferC** [comment]
Language changes maybe?

**Comment by JeffBezanson** [comment]
Standard library changes.

**Pull request merged**